### PR TITLE
[ISSUE-417] Fix CSI deployment via Operator on Openshift

### PR DESCRIPTION
**Pull request by safronovD**
### Issue dell/csi-baremetal#417

- increase memory limit
- fix port
- fix rbac

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [x] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Testing details_

### CODE REVIEW

File: charts/csi-baremetal-operator/templates/manager.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ spec:
         resources:
           limits:
             cpu: 100m
-            memory: 30Mi
+            memory: 100Mi
```
Comment (mishoyama) [inline charts/csi-baremetal-operator/templates/manager.yaml:32]: Why 30Mi is enough for RKE and vanilla K8s, but not for OpenShift? What is the difference?

File: charts/csi-baremetal-operator/templates/rbac.yaml [hunk 2]
```diff
@@ -38,6 +38,12 @@ rules:
   - "*"
   verbs:
   - "*"
+- apiGroups:
```
Comment (mishoyama) [inline charts/csi-baremetal-operator/templates/rbac.yaml:41]: Please add comments that these parameters required for OpenShift only. Don't we need to use platform flag here as well?

File: pkg/scheduler_patcher_openshift.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ const (
    "apiVersion" : "v1",
    "extenders": [
         {
-            "urlPrefix": "http://127.0.0.1:$PORT",
+            "urlPrefix": "http://127.0.0.1:8889",
```
Comment (mishoyama) [inline pkg/scheduler_patcher_openshift.go:24]: Port number should be in constants or passed via parameter ideally.

**Review by mishoyama — CHANGES_REQUESTED** [review]

File: charts/csi-baremetal-deployment/values.yaml [hunk 4]
```diff
@@ -124,6 +124,7 @@ scheduler:
     image:
       tag:
 
+    url_prefix: http://127.0.0.1:8889
```
Comment (mishoyama) [inline charts/csi-baremetal-deployment/values.yaml:127]: I suggest to pass port number only. Don't think that we need to set IP address in values as well.

File: charts/csi-baremetal-operator/templates/manager.yaml (hunk 1 above)
Comment (safronovD) [reply charts/csi-baremetal-operator/templates/manager.yaml:32]: It looks like that it's a bug in OpenShift 4.6:
- https://bugzilla.redhat.com/show_bug.cgi?id=1904558
- https://github.com/opencontainers/runc/pull/2812

File: charts/csi-baremetal-operator/templates/rbac.yaml (hunk 2 above)
Comment (safronovD) [reply charts/csi-baremetal-operator/templates/rbac.yaml:42]: Comment added. We use platform flag in csi-baremetal-deployment and it won't work for operator chart. Or we have to add that flag here too

File: charts/csi-baremetal-deployment/values.yaml (hunk 4 above)
Comment (safronovD) [reply charts/csi-baremetal-deployment/values.yaml]: Fixed. New parameter is called `extender_port`

File: pkg/scheduler_patcher_openshift.go (hunk 3 above)
Comment (safronovD) [reply pkg/scheduler_patcher_openshift.go]: fixed

File: charts/csi-baremetal-operator/templates/rbac.yaml (hunk 2 above)
Comment (mishoyama) [reply charts/csi-baremetal-operator/templates/rbac.yaml:42]: Ideally we need to pass _platform_ parameter to the operator. Operator will use it later when deploying CSI.

File: charts/csi-baremetal-operator/templates/manager.yaml (hunk 1 above)
Comment (mishoyama) [reply charts/csi-baremetal-operator/templates/manager.yaml:32]: Ok, please please comments in the code.

File: charts/csi-baremetal-deployment/values.yaml (hunk 4 above)
Comment (mishoyama) [reply charts/csi-baremetal-deployment/values.yaml]: Please use:
```
scheduler:
  extender:
    port: 8889
```

File: charts/csi-baremetal-operator/templates/rbac.yaml (hunk 2 above)
Comment (safronovD) [reply charts/csi-baremetal-operator/templates/rbac.yaml:42]: Also, we use platform parameter in `patcher-configmap`, which creates with Deployment chart. Should we move configmap deploying into operator code or into operator helm chart?

File: charts/csi-baremetal-deployment/values.yaml (hunk 4 above)
Comment (safronovD) [reply charts/csi-baremetal-deployment/values.yaml]: Fixed. It represents in deployment CRD as `extenderPort` to not create new data struct with one field

File: charts/csi-baremetal-operator/templates/manager.yaml (hunk 1 above)
Comment (safronovD) [reply charts/csi-baremetal-operator/templates/manager.yaml:33]: Added

File: charts/csi-baremetal-operator/templates/rbac.yaml (hunk 2 above)
Comment (mishoyama) [reply charts/csi-baremetal-operator/templates/rbac.yaml:42]: Could you please create issue for this and add TODO in the code to address in the next pull request?

**Review by mishoyama — APPROVED** [review]

File: charts/csi-baremetal-operator/templates/rbac.yaml (hunk 2 above)
Comment (safronovD) [reply charts/csi-baremetal-operator/templates/rbac.yaml:43]: TODO issue was added

**Pull request merged**